### ISSUE TITLE
freenas-api-iscsi does not delete zvols on Truenas

### ISSUE THREAD

**Issue by saro476** [democratic-csi/democratic-csi#417]
Hello,

If this issue has been mentioned before, feel free to point me to it, however I could not find anything related.

### Issue
freenas-api-iscsi drive successfully creates and on TrueNAS Scale, but upon deletion does not remove the volume. After a kubectl delete command on the pvc, the pvc is no longer in k8s resources, but it remains on the TrueNAS server.

After the kubectl delete command, I don't see any delete-like log entry from the csi-driver.

Any advice would be appreciated.

### Commands
Create claim
```
kubectl -n democratic-csi create \
-f test-claim-iscsi.yaml
```

Delete claim
```
kubectl -n democratic-csi delete \
-f test-claim-iscsi.yaml
```

### Logs, Files, and other system information
[democratic-csi-isci.log](https://github.com/user-attachments/files/16818652/democratic-csi-isci.log)
[democratic-csi-isci_csi-driver.log](https://github.com/user-attachments/files/16818653/democratic-csi-isci_csi-driver.log)
[test-claim-iscsi.txt](https://github.com/user-attachments/files/16818654/test-claim-iscsi.txt)
[values-iscsi.txt](https://github.com/user-attachments/files/16818655/values-iscsi.txt)
ISCSI Server: TrueNAS-SCALE-23.10.2
K8s Cluster: TalosOS v1.7.6

**Comment by jodoscott** [comment]
Can confirm the same for a small period of time (a few minutes), the zvol is still present after the volume is deleted, but a few minutes later it was cleaned up.

OS Version:TrueNAS-SCALE-23.10.2
K8s Cluster: TalosOS v1.7.6
Kubernetes Version: 1.30.4

**Comment by travisghansen** [comment]
Yeah, you just need to make sure the storage class has `Delete` set as the `ReclaimPolicy` and it should delete.
